icon.c

```diff
@@ -0,0 +1,161 @@
+#include <stdio.h>
+#include <assert.h>
+#include <cairo/cairo.h>
+#include <gdk-pixbuf/gdk-pixbuf.h>
+
+#include "icon.h"
+
+cairo_surface_t* gdk_cairo_image_surface_create_from_pixbuf(const GdkPixbuf *gdkbuf) {
```

Comment:
Hmm, I'm not a fan of this function.

Reply:
Name-wise or functionalify-wise? The name is unchanged from swaybg, and the function itself is the same apart from logging changes and some ifdefs.

Reply:
My bad. Apparently I based the name off of a PR and not the current code. I could make it more similar to https://github.com/swaywm/sway/blob/master/common/background-image.c if that helps?

Reply:
Is this different from `gdk_cairo_surface_create_from_pixbuf`? https://developer.gnome.org/gdk3/stable/gdk3-Cairo-Interaction.html#gdk-cairo-surface-create-from-pixbuf`

Reply:
Yeah, using `gdk_cairo_image_surface_create_from_pixbuf` would be a lot better :)

Reply:
There's also `gdk_cairo_set_source_pixbuf` which I use in oguri. I'm not familiar enough to know what the particular upsides and downsides of each method are.

Reply:
I guess the downside of both those is a dependence also on gdk (although that can be made optional along with gdk-pixbuf). For the latter I assume that would be used in `render.c`? If so it might be nicer to do the former (assuming it works properly) and return a cairo surface as I do now. That way all the ifdefs related to optional gdk stuff is constrained to `icon.c`.